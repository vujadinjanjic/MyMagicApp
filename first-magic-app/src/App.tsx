import React from "react";
import "./App.css";
import Homepage from "./modules/Homepage/Homepage";
import { BrowserRouter as Router, Switch, Route } from "react-router-dom";
import Cardpage from "./modules/Cardpage/Cardpage";

function App() {
  return (
    <div className="App">
      <Router>
        <Switch>
          <Route path="/">
            <Homepage />
          </Route>
          <Route path="/card/:id">
            <Cardpage />
          </Route>
        </Switch>
      </Router>
    </div>
  );
}

export default App;
