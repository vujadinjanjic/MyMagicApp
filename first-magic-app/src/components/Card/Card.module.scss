.card {
  background: rgba(#ffffff, 0.6);
  display: flex;
  flex-direction: column;
  margin: 10px 10px;
  position: relative;
  overflow: hidden;
  width: auto;
  box-shadow: 0px 4px 16px rgba(#000000, 0.05);
  border-radius: 12px;
  transition: 150ms ease;
  &:hover {
    box-shadow: 0px 4px 12px rgba(#000000, 0.2);
  }
  cursor: pointer;
}
.magicImage {
  max-height: 120px;
  min-height: 120px;
  height: 100%;
  width: 100%;
  object-fit: cover;
}

.bottom {
  padding: 30px 24px;
  overflow: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
}

.button {
  color: green;
  border-color: green;
  cursor: pointer;
  border-radius: 40px;
  font-weight: 400;
  font-size: 12px;
  padding: 1px 1px;
  padding: 5px 20px;
  margin-top: 20px;
}

.cardName {
  color: grey;
  font-size: 20px;
  font-weight: 700;
}

.buttonsContainer {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  margin-top: 20px;
}
.icon {
  width: 25px;
  height: 25px;
  cursor: pointer;
}
