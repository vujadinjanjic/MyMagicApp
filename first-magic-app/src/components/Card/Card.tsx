import React from "react";
import magicImage from "../../images/trick.png";
import styles from "./Card.module.scss";
import { useHistory } from "react-router";
import redHeart from "../../images/red-heart.svg";
import emptyHeart from "../../images/empty-heart.svg";

interface ICardProps {
  name: string;
  id: string;
  added?: boolean;
}

const Card = ({ name, id, added }: ICardProps) => {
  const navigateTo = useHistory();
  return (
    <div
      className={styles.card}
      key={id}
      onClick={() => navigateTo.push(`/card/${id?.split("/")[3]}`)}
    >
      <img className={styles.magicImage} src={magicImage} />
      <div className={styles.bottom}>
        <span className={styles.cardName} aria-label="name">
          {name}
        </span>
        {added ? (
          <img src={redHeart} className={styles.icon} aria-label="fullFilled" />
        ) : (
          <img src={emptyHeart} className={styles.icon} aria-label="empty" />
        )}
      </div>
    </div>
  );
};

export default Card;
