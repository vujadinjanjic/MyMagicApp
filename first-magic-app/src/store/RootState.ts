import { SpellState } from "./spell/types";

export interface RootState {
  spell: SpellState;
}
