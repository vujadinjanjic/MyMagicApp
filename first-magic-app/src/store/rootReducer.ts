import { combineReducers } from "redux";
import spellReducer from "./spell/reducer";

export default combineReducers({
  spell: spellReducer,
});
