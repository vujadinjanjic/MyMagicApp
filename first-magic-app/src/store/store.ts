import { createStore } from "redux";
import rootReducer from "./rootReducer";
import { composeWithDevTools } from "redux-devtools-extension";
import { persistReducer } from "redux-persist";
import storage from "redux-persist/lib/storage";

const persistConfig = {
  key: "root",
  storage,
};

const persistedReducer = persistReducer(persistConfig, rootReducer);
const store = createStore(persistedReducer, composeWithDevTools());
export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;

export default store;
