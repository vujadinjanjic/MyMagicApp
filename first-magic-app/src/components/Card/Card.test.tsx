import { fireEvent, render as rtlRender, screen } from "@testing-library/react";
import store from "../../store/store";
import { Provider } from "react-redux";
import userEvent from "@testing-library/user-event";
import renderer from "react-test-renderer";
import Card from "./Card";
import { createMemoryHistory } from "history";
import { Route, Router } from "react-router";
import App from "../../App";

const mockedProps = {
  name: "name",
  id: "1",
  added: true,
};

const render = (component: any) =>
  rtlRender(<Provider store={store}>{component}</Provider>);

describe("Card testing", () => {
  it("renders correctly", async () => {
    const tree = renderer.create(<Card {...mockedProps} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("display name", async () => {
    render(<Card {...mockedProps} />);
    const name = await screen.findByLabelText("name");
    expect(name).toBeInTheDocument();
  });

  it("display fullfiled heart when magic is in collection", async () => {
    render(<Card {...mockedProps} />);
    const img = await screen.findByLabelText("fullFilled");
    expect(img).toBeInTheDocument();
  });

  it("display empty heart when magic is not collected", async () => {
    render(<Card {...mockedProps} added={false} />);
    const img = await screen.findByLabelText("empty");
    expect(img).toBeInTheDocument();
  });
});
