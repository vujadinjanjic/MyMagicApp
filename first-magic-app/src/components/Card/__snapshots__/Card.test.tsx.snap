// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card testing renders correctly 1`] = `
<div
  className="card"
  onClick={[Function]}
>
  <img
    className="magicImage"
    src="trick.png"
  />
  <div
    className="bottom"
  >
    <span
      aria-label="name"
      className="cardName"
    >
      name
    </span>
    <img
      aria-label="fullFilled"
      className="icon"
      src="red-heart.svg"
    />
  </div>
</div>
`;
